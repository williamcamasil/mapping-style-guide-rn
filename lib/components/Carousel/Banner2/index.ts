export default require('./index.png');
