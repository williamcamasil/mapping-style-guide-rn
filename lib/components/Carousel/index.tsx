import React, {
  ReactNode,
  useCallback,
  useState,
  useImperativeHandle,
  useRef,
  useEffect,
  RefObject,
} from 'react';
import {
  FlatList,
  StyleSheet,
  View,
  LayoutChangeEvent,
  LayoutRectangle,
  ViewToken,
  StyleProp,
  ViewStyle,
  ListRenderItemInfo,
} from 'react-native';

import { Images } from '../../assets';
import { useViewStyles } from '../../hooks';
import { AppThemeType, DefaultTheme, withTheme } from '../../theme';
import Ratio from '../Ratio';
import Spacer from '../Spacer';

export type CarouselRefType = {
  goToNext: () => void;
  goToPrevious: () => void;
  goToIndex: (index: number) => void;
  flatListRef: RefObject<FlatList>;
};

export type CarouselPropsType = {
  data: ReactNode[];
  innerRef?: React.MutableRefObject<CarouselRefType | null>;
  onItemChange?: (index: number) => void;
  theme: AppThemeType;
  style?: StyleProp<ViewStyle>;
  pointersContainerStyle?: StyleProp<ViewStyle>;
  autoPlayIntervalMilliseconds?: number;
  autoPlay?: boolean;
  originalWidth: number;
  originalHeight: number;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  containerPointer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

const viewabilityConfig = {
  waitForInteraction: false,
  itemVisiblePercentThreshold: 50,
};

const Carousel: React.FC<CarouselPropsType> = ({
  data, theme, onItemChange, innerRef, style,
  pointersContainerStyle,
  autoPlayIntervalMilliseconds: intervalToTimeoutMilliseconds,
  autoPlay: useTimer,
  originalWidth, originalHeight,
}) => {
  const flatListRef = useRef<FlatList>(null);

  const [itemSelected, setItemSelected] = useState(0);
  const [listLayout, setListLayout] = useState<LayoutRectangle | undefined>();

  const getIndexToScroll = useCallback((index: number): number => {
    if (index === -1) {
      return data.length - 1;
    }

    if (index === data.length) {
      return 0;
    }

    return index;
  }, [data.length]);

  const scrollToIndex = useCallback((index: number) => {

    const indexToScroll = getIndexToScroll(index);

    flatListRef.current?.scrollToIndex({
      index: indexToScroll,
      animated: true,
    });
  }, [getIndexToScroll]);

  useImperativeHandle(
    innerRef,
    () => ({
      goToNext: () => {
        scrollToIndex(itemSelected + 1);
      },
      goToPrevious: () => {
        scrollToIndex(itemSelected - 1);
      },
      goToIndex: scrollToIndex,
      flatListRef,
    }),
    [scrollToIndex, itemSelected],
  );

  useEffect(() => {
    if (!useTimer) return undefined;

    const timeoutInstance = setTimeout(() => {
      scrollToIndex(itemSelected + 1);
    }, intervalToTimeoutMilliseconds);

    return () => clearTimeout(timeoutInstance);
  }, [intervalToTimeoutMilliseconds, itemSelected, scrollToIndex, useTimer]);

  const itemContainerStyles = useViewStyles(
    () => [
      {
        width: listLayout?.width,
        height: listLayout?.height,
        overflow: 'hidden',
      },
    ],
    [listLayout],
  );

  const containerStyles = useViewStyles(() => [
    styles.container,
    style,
  ], [style]);

  const containerPointerStyles = useViewStyles(() => [
    styles.containerPointer,
    pointersContainerStyle,
  ], [pointersContainerStyle]);

  const pointerStyles = useViewStyles(() => [
    {
      marginHorizontal: theme.spacings.sNano,
    },
  ], [theme.spacings.sNano]);

  const renderItem = useCallback((item: ListRenderItemInfo<ReactNode>) => <View testID="item-container-carousel" style={itemContainerStyles}>{item.item}</View>, [itemContainerStyles]);

  const onViewableItemsChanged = useCallback(
    (info: { viewableItems: Array<ViewToken>; changed: Array<ViewToken> }) => {

      const { viewableItems } = info;
      const currentIndex = viewableItems?.[0]?.index;

      if (typeof currentIndex !== 'number') return;

      setItemSelected(currentIndex);
      onItemChange?.(currentIndex);
    },
    [onItemChange],
  );

  const handleListLayout = useCallback((event: LayoutChangeEvent) => {
    setListLayout(event.nativeEvent.layout);
  }, []);

  const getItemLayout = useCallback(
    (item: unknown, index: number): { length: number; offset: number; index: number } => {

      const length = listLayout?.width ?? 0;
      return {
        index,
        offset: length * index,
        length,
      };
    },
    [listLayout],
  );

  const getColorPointer = (index: number) => {
    if (itemSelected === index) return theme.colors.primary200;

    return theme.colors.neutralGray300;
  };

  const getFillPointer = (index: number) => {
    if (itemSelected === index) { return theme.colors.primaryMain; }

    return undefined;
  };

  const renderPointerSelected = () => (
    <View testID="container-pointers" style={containerPointerStyles}>
      {data.map((item, index) => {
        const key = `pointer-${index}`;

        return (
          <Images.Pointer
            testID="item-pointer"
            key={key}
            color={getColorPointer(index)}
            fill={getFillPointer(index)}
            style={pointerStyles}
          />
        );
      })}
    </View>
  );

  return (
    <View style={containerStyles}>
      <Ratio
        originalWidth={originalWidth}
        originalHeight={originalHeight}
      >
        <FlatList
          testID="carousel-flat-list-id"
          ref={flatListRef}
          onLayout={handleListLayout}
          data={data}
          renderItem={renderItem}
          getItemLayout={getItemLayout}
          horizontal
          pagingEnabled
          showsHorizontalScrollIndicator={false}
          onViewableItemsChanged={onViewableItemsChanged}
          viewabilityConfig={viewabilityConfig}
        />
      </Ratio>
      <Spacer size={DefaultTheme.spacings.sLarge} />
      {renderPointerSelected()}
    </View>
  );
};

Carousel.defaultProps = {
  autoPlay: true,
  autoPlayIntervalMilliseconds: 4000,
  originalWidth: 375,
  originalHeight: 384,
};

export default withTheme(Carousel);
