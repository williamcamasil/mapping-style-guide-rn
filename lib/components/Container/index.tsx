import React from 'react';
import { View, ViewProps, ViewStyle } from 'react-native';

import { useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';

export type ContainerPropsType = ViewProps & {
  theme: AppThemeType;
  padding?: ViewStyle['padding'];
  paddingVertical?: ViewStyle['padding'];
  paddingHorizontal?: ViewStyle['padding'];
  paddingLeft?: ViewStyle['padding'];
  paddingRight?: ViewStyle['padding'];
  paddingTop?: ViewStyle['padding'];
  paddingBottom?: ViewStyle['padding'];
  fillParent?: boolean;
  horizontal?: boolean;
  centered?: boolean;
  backgroundColor?: ViewStyle['backgroundColor'];
  blockActions?: boolean;
  justifyContent?: ViewStyle['justifyContent'];
  alignItems?: ViewStyle['alignItems'];
};

const Container: React.FC<ContainerPropsType> = ({
  children,
  style,
  theme,
  padding,
  paddingVertical,
  paddingHorizontal,
  paddingLeft,
  paddingRight,
  paddingTop,
  paddingBottom,
  fillParent,
  horizontal,
  centered,
  backgroundColor,
  blockActions,
  justifyContent,
  alignItems,
  ...others
}) => {
  const containerStyles = useViewStyles(() => [
    {
      flex: fillParent ? 1 : undefined,
      flexDirection: horizontal ? 'row' : undefined,
      justifyContent: justifyContent ?? (centered && horizontal ? 'center' : undefined),
      alignItems: alignItems ?? (centered && !horizontal ? 'center' : undefined),
      backgroundColor: backgroundColor || theme.colors.neutralWhite,
      padding,
      paddingVertical,
      paddingHorizontal,
      paddingLeft,
      paddingRight,
      paddingTop,
      paddingBottom,
    },
    style,
  ], [
    backgroundColor,
    centered,
    fillParent,
    horizontal,
    padding,
    paddingBottom,
    paddingHorizontal,
    paddingLeft,
    paddingRight,
    paddingTop,
    paddingVertical,
    style,
    theme.colors.neutralWhite,
    justifyContent,
    alignItems,
  ]);

  return (
    <View pointerEvents={blockActions ? 'none' : 'auto'} style={containerStyles} {...others}>
      {children}
    </View>
  );
};

Container.defaultProps = {
  fillParent: true,
};

export default withTheme(Container);
