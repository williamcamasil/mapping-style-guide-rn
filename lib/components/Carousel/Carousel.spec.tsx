import React, { createRef, ReactNode } from 'react';
import { Image } from 'react-native';

import {
  render, fireEvent, waitFor, act,
} from '@testing-library/react-native';

import Carousel, { CarouselRefType } from '.';
import { DefaultTheme } from '../../theme';
import Container from '../Container';
import Banner1 from './Banner1';
import Banner2 from './Banner2';
import Banner3 from './Banner3';

const data: ReactNode[] = [
  <Container centered>
    <Image
      testID="image-banner1"
      source={Banner1}
      style={{
        width: '100%',
      }}
    />
  </Container>,
  <Container centered>
    <Image
      testID="image-banner2"
      source={Banner2}
      style={{
        width: '100%',
      }}
    />
  </Container>,
  <Container centered>
    <Image
      testID="image-banner3"
      source={Banner3}
      style={{
        width: '100%',
      }}
    />
  </Container>,
];

describe('Carousel', () => {
  it('Carousel snapshot', () => {
    const carouselRef = createRef<CarouselRefType>();

    const result = render(
      <Carousel
        innerRef={carouselRef}
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        autoPlay={false}
        originalWidth={375}
        originalHeight={384}
      />,
    ).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('Should render the banners', () => {
    const result = render(
      <Carousel
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        autoPlay={false}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    result.getByTestId('image-banner1');
    result.getByTestId('image-banner2');
    result.getByTestId('image-banner3');
  });

  it('Should render one pointer per element within data totaling 3 rendered pointers', () => {
    const result = render(
      <Carousel
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        autoPlay={false}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    const pointers = result.getAllByTestId('item-pointer');

    expect(pointers.length).toBe(3);
  });

  it(`Should call function goToNext in ref to change banner and modify pointer checked position 
  for the next`, async () => {
    const carouselRef = createRef<CarouselRefType>();

    const result = render(
      <Carousel
        innerRef={carouselRef}
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        autoPlay={false}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    await waitFor(() => {
      expect(carouselRef.current?.flatListRef.current).toBeDefined();
    });

    act(() => carouselRef.current?.goToNext());

    fireEvent(result.getByTestId('carousel-flat-list-id'), 'onViewableItemsChanged', {
      viewableItems: [{
        index: 1,
      }],
      changed: [],
    });

    const containerPointers = result.getByTestId('container-pointers');

    expect(containerPointers.props.children[0].props).toHaveProperty('color', DefaultTheme.colors.neutralGray300);
    expect(containerPointers.props.children[0].props).toHaveProperty('fill', undefined);

    expect(containerPointers.props.children[1].props).toHaveProperty('color', DefaultTheme.colors.primary200);
    expect(containerPointers.props.children[1].props).toHaveProperty('fill', DefaultTheme.colors.primaryMain);

  });

  it(`Should call the goToPrevious function in ref to change the banner and modify the position of the
pointer to previous`, async () => {
    const carouselRef = createRef<CarouselRefType>();

    const result = render(
      <Carousel
        innerRef={carouselRef}
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        autoPlay={false}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    await waitFor(() => {
      expect(carouselRef.current?.flatListRef.current).toBeDefined();
    });

    act(() => carouselRef.current?.goToNext());
    act(() => carouselRef.current?.goToPrevious());

    const containerPointers = result.getByTestId('container-pointers');

    expect(containerPointers.props.children[1].props).toHaveProperty('color', DefaultTheme.colors.neutralGray300);
    expect(containerPointers.props.children[1].props).toHaveProperty('fill', undefined);

    expect(containerPointers.props.children[0].props).toHaveProperty('color', DefaultTheme.colors.primary200);
    expect(containerPointers.props.children[0].props).toHaveProperty('fill', DefaultTheme.colors.primaryMain);
  });

  it('Should call the goToNext function passing from the last element to the first', async () => {
    const carouselRef = createRef<CarouselRefType>();

    const result = render(
      <Carousel
        innerRef={carouselRef}
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        autoPlay={false}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    fireEvent(result.getByTestId('carousel-flat-list-id'), 'onViewableItemsChanged', {
      viewableItems: [{
        index: 2,
      }],
      changed: [],
    });

    await waitFor(() => {
      expect(carouselRef.current?.flatListRef.current).toBeDefined();
    });

    act(() => carouselRef.current?.goToNext());

    fireEvent(result.getByTestId('carousel-flat-list-id'), 'onViewableItemsChanged', {
      viewableItems: [{
        index: 0,
      }],
      changed: [],
    });

    const containerPointers = result.getByTestId('container-pointers');

    expect(containerPointers.props.children[2].props).toHaveProperty('color', DefaultTheme.colors.neutralGray300);
    expect(containerPointers.props.children[2].props).toHaveProperty('fill', undefined);

    expect(containerPointers.props.children[0].props).toHaveProperty('color', DefaultTheme.colors.primary200);
    expect(containerPointers.props.children[0].props).toHaveProperty('fill', DefaultTheme.colors.primaryMain);
  });

  it('Should call the goToPrevios function on the first element to go to the last element', async () => {
    const carouselRef = createRef<CarouselRefType>();

    const result = render(
      <Carousel
        innerRef={carouselRef}
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    act(() => carouselRef.current?.goToPrevious());

    fireEvent(result.getByTestId('carousel-flat-list-id'), 'onViewableItemsChanged', {
      viewableItems: [{
        index: 2,
      }],
      changed: [],
    });

    const containerPointers = result.getByTestId('container-pointers');

    expect(containerPointers.props.children[0].props).toHaveProperty('color', DefaultTheme.colors.neutralGray300);
    expect(containerPointers.props.children[0].props).toHaveProperty('fill', undefined);

    expect(containerPointers.props.children[2].props).toHaveProperty('color', DefaultTheme.colors.primary200);
    expect(containerPointers.props.children[2].props).toHaveProperty('fill', DefaultTheme.colors.primaryMain);
  });

  it('Should call event onViewableItemsChanged and call method onItemChange with index value 1', async () => {
    const carouselRef = createRef<CarouselRefType>();
    const onItemChange = jest.fn();

    const result = render(
      <Carousel
        innerRef={carouselRef}
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        onItemChange={onItemChange}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    fireEvent(result.getByTestId('carousel-flat-list-id'), 'onViewableItemsChanged', {
      viewableItems: [{
        index: 1,
      }],
      changed: [],
    });

    await waitFor(() => expect(onItemChange).toHaveBeenCalledWith(1));
  });

  it('Should call onViewableItemsChanged event and pass viewableItems as undefined not calling onItemChange function', async () => {
    const carouselRef = createRef<CarouselRefType>();
    const onItemChange = jest.fn();

    const result = render(
      <Carousel
        innerRef={carouselRef}
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        onItemChange={onItemChange}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    fireEvent(result.getByTestId('carousel-flat-list-id'), 'onViewableItemsChanged', {
      viewableItems: null,
      changed: [],
    });

    await waitFor(() => expect(onItemChange).not.toHaveBeenCalled());
  });

  it('Should call event onLayout to set values to listLayout', () => {
    const result = render(
      <Carousel
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    fireEvent(result.getByTestId('carousel-flat-list-id'), 'layout', {
      nativeEvent: { layout: { width: 300, height: 450 } },
    });

    const itemContainer = result.getAllByTestId('item-container-carousel');

    expect(itemContainer[0].props.style).toHaveProperty('width', 300);
    expect(itemContainer[0].props.style).toHaveProperty('height', 450);
  });

  it('Should call the function scrollToIndex in the period set in intervalToTimeoutMilliseconds', async () => {
    const onItemChange = jest.fn();

    const carouselRef = createRef<CarouselRefType>();

    render(
      <Carousel
        innerRef={carouselRef}
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        onItemChange={onItemChange}
        autoPlayIntervalMilliseconds={10}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    await waitFor(() => {
      expect(carouselRef.current?.flatListRef.current).toBeDefined();
    });

    const spyScrollToIndex = jest.spyOn(carouselRef.current!.flatListRef.current!, 'scrollToIndex');

    await waitFor(() => {
      expect(spyScrollToIndex).toHaveBeenCalledWith({ animated: true, index: 1 });
    });
  });

  it('Should not call the function scrollToIndex after set useTimer to false', async () => {
    const onItemChange = jest.fn();

    const carouselRef = createRef<CarouselRefType>();

    render(
      <Carousel
        innerRef={carouselRef}
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        onItemChange={onItemChange}
        autoPlay={false}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    await waitFor(() => {
      expect(carouselRef.current?.flatListRef.current).toBeDefined();
    });

    const spyScrollToIndex = jest.spyOn(carouselRef.current!.flatListRef.current!, 'scrollToIndex');

    expect(spyScrollToIndex).not.toHaveBeenCalled();
  });

  it('Should call the function scrollToIndex after the default time of 4s', async () => {
    const onItemChange = jest.fn();

    const carouselRef = createRef<CarouselRefType>();

    render(
      <Carousel
        innerRef={carouselRef}
        data={data}
        style={{ height: 475, flex: 0 }}
        pointersContainerStyle={{ marginLeft: 20, justifyContent: 'flex-start' }}
        onItemChange={onItemChange}
        originalWidth={375}
        originalHeight={384}
      />,
    );

    await waitFor(() => {
      expect(carouselRef.current?.flatListRef.current).toBeDefined();
    });

    const spyScrollToIndex = jest.spyOn(carouselRef.current!.flatListRef.current!, 'scrollToIndex');

    await waitFor(() => {
      expect(spyScrollToIndex).toHaveBeenCalled();
    }, { timeout: 5000 });
  });
});
