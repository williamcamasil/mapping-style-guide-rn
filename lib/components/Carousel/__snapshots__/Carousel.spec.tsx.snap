// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Carousel snapshot 1`] = `
<View
  style={
    {
      "flex": 0,
      "height": 475,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        data={
          [
            <withTheme(Container)
              centered={true}
            >
              <Image
                source={
                  {
                    "testUri": "../../../lib/components/Carousel/Banner1/index.png",
                  }
                }
                style={
                  {
                    "width": "100%",
                  }
                }
                testID="image-banner1"
              />
            </withTheme(Container)>,
            <withTheme(Container)
              centered={true}
            >
              <Image
                source={
                  {
                    "testUri": "../../../lib/components/Carousel/Banner2/index.png",
                  }
                }
                style={
                  {
                    "width": "100%",
                  }
                }
                testID="image-banner2"
              />
            </withTheme(Container)>,
            <withTheme(Container)
              centered={true}
            >
              <Image
                source={
                  {
                    "testUri": "../../../lib/components/Carousel/Banner3/index.png",
                  }
                }
                style={
                  {
                    "width": "100%",
                  }
                }
                testID="image-banner3"
              />
            </withTheme(Container)>,
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        getItemLayout={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onViewableItemsChanged={[Function]}
        pagingEnabled={true}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={[]}
        testID="carousel-flat-list-id"
        viewabilityConfig={
          {
            "itemVisiblePercentThreshold": 50,
            "waitForInteraction": false,
          }
        }
        viewabilityConfigCallbackPairs={
          [
            {
              "onViewableItemsChanged": [Function],
              "viewabilityConfig": {
                "itemVisiblePercentThreshold": 50,
                "waitForInteraction": false,
              },
            },
          ]
        }
      >
        <View>
          <View
            onFocusCapture={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "height": undefined,
                  "overflow": "hidden",
                  "width": undefined,
                }
              }
              testID="item-container-carousel"
            >
              <View
                pointerEvents="auto"
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "flex": 1,
                    "flexDirection": undefined,
                    "justifyContent": undefined,
                    "padding": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "paddingVertical": undefined,
                  }
                }
              >
                <Image
                  source={
                    {
                      "testUri": "../../../lib/components/Carousel/Banner1/index.png",
                    }
                  }
                  style={
                    {
                      "width": "100%",
                    }
                  }
                  testID="image-banner1"
                />
              </View>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "height": undefined,
                  "overflow": "hidden",
                  "width": undefined,
                }
              }
              testID="item-container-carousel"
            >
              <View
                pointerEvents="auto"
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "flex": 1,
                    "flexDirection": undefined,
                    "justifyContent": undefined,
                    "padding": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "paddingVertical": undefined,
                  }
                }
              >
                <Image
                  source={
                    {
                      "testUri": "../../../lib/components/Carousel/Banner2/index.png",
                    }
                  }
                  style={
                    {
                      "width": "100%",
                    }
                  }
                  testID="image-banner2"
                />
              </View>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "height": undefined,
                  "overflow": "hidden",
                  "width": undefined,
                }
              }
              testID="item-container-carousel"
            >
              <View
                pointerEvents="auto"
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "flex": 1,
                    "flexDirection": undefined,
                    "justifyContent": undefined,
                    "padding": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "paddingVertical": undefined,
                  }
                }
              >
                <Image
                  source={
                    {
                      "testUri": "../../../lib/components/Carousel/Banner3/index.png",
                    }
                  }
                  style={
                    {
                      "width": "100%",
                    }
                  }
                  testID="image-banner3"
                />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <Text
    style={
      {
        "height": 28,
        "width": 28,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginLeft": 20,
      }
    }
    testID="container-pointers"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      fill="none"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "marginHorizontal": 8,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      testID="item-pointer"
      vbHeight={16}
      vbWidth={16}
      width={16}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGCircle
          cx="8"
          cy="8"
          fill={
            {
              "payload": 4293482516,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          r="7"
          stroke={
            {
              "payload": 4294831075,
              "type": 0,
            }
          }
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      fill="none"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "marginHorizontal": 8,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      testID="item-pointer"
      vbHeight={16}
      vbWidth={16}
      width={16}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGCircle
          cx="8"
          cy="8"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
            ]
          }
          r="7"
          stroke={
            {
              "payload": 4292993764,
              "type": 0,
            }
          }
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      fill="none"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "marginHorizontal": 8,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      testID="item-pointer"
      vbHeight={16}
      vbWidth={16}
      width={16}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGCircle
          cx="8"
          cy="8"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
            ]
          }
          r="7"
          stroke={
            {
              "payload": 4292993764,
              "type": 0,
            }
          }
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
